src/timeMachine/components/Vis.tsx

```diff
@@ -71,6 +77,26 @@ const TimeMachineVis: FC<Props> = ({
     'time-machine--view__empty': noQueries,
   })
 
+  // Handles deadman check edge case to allow non-numeric values
+  if (
```

Comment:
i feel like this further accentuates a weird relationship between these components (vis and whatever middle bar), but idk what to do to smooth it over. Is this a sign of DRY breaking?

Reply:
I'm sure true DRY believers would tell us that we could decompose TimeMachine more to provide more abstraction layers that would ultimately remove some of these `if this then that` statements. But good lord I find myself questioning if adding that complexity is worth the investment. 

I'm a fan of KISS - keep it simple, stupid. 😆 A little code repetition, while it does potentially increase maintenance costs (e.g. fix a bug in two places), might be worth it if you're going to reduce the overall complexity of the code since less complexity means fewer bugs. 

So yes, I don't like this solution but de-coupling time machine sounds like a massive undertaking that we should strategize on....hopefully sooner rather than later 😅 

</end rant>